llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -389,12 +403,16 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
             self._save_checkpoint(
                 state,
                 logger,
-                register_to_mlflow=(
-                    self.mlflow_registered_model_name is not None and
-                    is_last_batch
+                log_to_mlflow=(
+                    self.save_folder is None or (
```

Comment:
we probably need some check somewhere that if save_folder is None you actually have an mlflow logger to use?

Reply:
We do a check on init that  errors if save_folder is None and mlflow_registered_model_name is None, so this should always be valid.

Reply:
actually I guess maybe we shouldn't require users to input a `mlflow_registered_model_name` for logging only

Reply:
updated